pkg/providers/interface.go

```diff
@@ -0,0 +1,10 @@
+package providers
+
+import (
+	pTypes "github.com/konveyor/asset-generation/pkg/providers/types/provider"
+)
+
+type Provider interface {
```

Comment:
What do you think to rename this interface to `Discoverer` instead of `Provider`?

I plan to create a new interface called `Generator` that exposes so far 1 function that is `Generate()`.

It's just to have a common naming convention because `Provider` is too ambiguous knowing that we have 2 types of providers:
* Discoverers
* Generators

I'm open to suggestions if you feel differently 😄